pootle/apps/pootle_app/management/commands/set_filetype.py

```diff
@@ -0,0 +1,68 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright (C) Pootle contributors.
+#
+# This file is a part of the Pootle project. It is distributed under the GPL3
+# or later license. See the LICENSE file for a copy of the license and the
+# AUTHORS file for copyright and authorship information.
+
+import os
+
+os.environ['DJANGO_SETTINGS_MODULE'] = 'pootle.settings'
+
+from django.core.management.base import CommandError
+
+from pootle_format.models import Format
+from pootle_project.models import Project
+
+from . import PootleCommand
+
+
+class Command(PootleCommand):
+    help = "Manage Store formats."
+
+    def add_arguments(self, parser):
+        super(Command, self).add_arguments(parser)
+        parser.add_argument(
+            'filetype',
+            action='store',
+            help="Format to set")
+        parser.add_argument(
+            '--from-filetype',
+            action='store',
+            help="Only convert Stores of this file type")
+        parser.add_argument(
+            '--matching',
+            action='store',
+            help="Glob match Store path relative excluding extension")
+
+    def handle_all(self, **options):
+        projects = []
+        try:
+            filetype = Format.objects.get(name=options["filetype"])
+        except Format.DoesNotExist:
+            raise CommandError("Unrecognized filetype '%s'" % options["filetype"])
+        if options["from_filetype"]:
+            try:
+                from_filetype = Format.objects.get(name=options["from_filetype"])
+            except Format.DoesNotExist:
+                raise CommandError(
+                    "Unrecognized filetype '%s'" % options["from_filetype"])
```

Comment:
Nit: move this retrieve filetype_or_fail logic to a separate method because it will help reading this method and will reduce code duplication.